incydr/cli/cmds/watchlists.py

```diff
@@ -0,0 +1,690 @@
+import click
+from click import Context
+from rich.panel import Panel
+from rich.progress import track
+
+from incydr._watchlists.models.responses import IncludedDepartment
+from incydr._watchlists.models.responses import IncludedDirectoryGroup
+from incydr._watchlists.models.responses import Watchlist
+from incydr._watchlists.models.responses import WatchlistUser
+from incydr.cli import console
+from incydr.cli import init_client
+from incydr.cli import log_file_option
+from incydr.cli import log_level_option
+from incydr.cli import render
+from incydr.cli.cmds.options.output_options import columns_option
+from incydr.cli.cmds.options.output_options import single_format_option
+from incydr.cli.cmds.options.output_options import SingleFormat
+from incydr.cli.cmds.options.output_options import table_format_option
+from incydr.cli.cmds.options.output_options import TableFormat
+from incydr.cli.cmds.options.utils import user_lookup_callback
+from incydr.cli.cmds.utils import user_lookup
+from incydr.cli.core import incompatible_with
+from incydr.cli.core import IncydrCommand
+from incydr.cli.core import IncydrGroup
+from incydr.enums.watchlists import WatchlistType
+from incydr.utils import model_as_card
+from incydr.utils import read_dict_from_csv
+
+
+# TODO: check for custom title?
+def get_watchlist_id_callback(ctx, param, value):
```

Comment:
This callback that I'm using for every `watchlist` arg currently only takes a watchlist type OR a watchlist ID.  

I don't currently allow for passing the title of a `custom` watchlist because I didn't see a way to determine if something was a `title` or an `ID` without making an API call for every value passed even if it was an ID (to see if it matched any existing watchlist titles).  

That functionality is in the SDK but I'm wondering if that should be removed because I'm not sure if watchlists are required to have unique titles?

Reply:
watchlist_ids are UUIDs, so we can attempt to parse it to determine if it's an ID or name: 

```
>>> from uuid import UUID
>>> UUID(hex="1c7dd799-1aa0-4f3a-bae8-1d3242fc2af6")
UUID('1c7dd799-1aa0-4f3a-bae8-1d3242fc2af6')
>>> UUID(hex="non-uid")
Traceback (most recent call last):
  File "<input>", line 1, in <module>
    UUID(hex="non-uid")
  File "/opt/homebrew/Cellar/python@3.10/3.10.6_2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/uuid.py", line 177, in __init__
    raise ValueError('badly formed hexadecimal UUID string')
ValueError: badly formed hexadecimal UUID string
```

Reply:
Oh very nice, should we just assume watchlists will have unique titles?

Reply:
Just found out click also has a `click.UUID` parameter type if we ever have a field that we only want to accept uuids 👀    https://click.palletsprojects.com/en/7.x/api/#click.UUID